utils/build-apple-framework.sh

```diff
@@ -56,11 +59,16 @@ function configure_apple_framework {
   else
     build_cli_tools="false"
   fi
+  if [[ $1 == catalyst ]]; then
+    catalyst="true"
+    platform="macosx"
+  fi
 
   local cmake_flags=" \
-    -DHERMES_APPLE_TARGET_PLATFORM:STRING=$1 \
```

Comment:
Please keep in mind that `$1` is still referred to on line 83. You probably want this to be replaced. If you don't do this, you are not building "build_macosx", but "build_catalyst" and effectively, always include Mac artifacts in your `xcframework`.

Reply:
**IMPORTANT**: However, if you read my comment on the caching/producing MacOS / Catalyst frameworks at the same time [here](https://github.com/facebook/hermes/pull/475/files/b63d2408861671d5e3eb4f7697647cd04d3501c7#r621143952), this could actually stay that way, since we want a different build folder for Mac and different build folder for Catalyst.